### DISCUSSION TITLE
C/C++ Extension Pack

### DISCUSSION THREAD

**Discussion by Zingam** [microsoft/vscode-cpptools#9939] [Q&A]
I remember back in the day the C/C++ Extension Pack included a bunch of extensions. Now they are just:

C++ Tools
C++ Themes
CMake
CMake Tools

I'm looking at my installed extensions and I see extensions like Better C++ Syntax, Doxygen, which I think were installed by the extension pack. Are these extra extensions still needed? Is their functionality covered by the core C++ Tools extension these days?

I use a recommended.json and I push it to the git repo. I wonder if I need to clean up my list based on latest C/C++ Extension Pack.

**Comment by sean-mcmanus** [comment]
We trimmed it down to only the most essential extensions. "Better C++ Syntax" is optional, since the changes in that extension eventually get merged into VS Code -- it may have newer changes/fixes. The Doxygen one is optional too, we've added support for some of the features it provided, but you can still install it if you want some additional features we don't provide yet. You could still include those extensions in your recommendations if you want.

**Reply by KaeLL** [reply]
Will "C/C++ Themes" be merged into cpptools? I don't even get why it's a separate extension.

**Reply by sean-mcmanus** [reply]
@KaeLL I recall we encountered a reason to not hard-bundle it into the C/C++ extension -- it is soft-bundled in the "C/C++ Extension Pack" extension. @Colengms might remember what the reason was. Also, VS Code theme extensions in general also usually separate from others.

**Reply by sean-mcmanus** [reply]
@KaeLL Found the reason -- we didn't want this to happen when the C/C++ extension was installed:
![image](https://user-images.githubusercontent.com/19859882/194092536-4356817c-4833-4de8-b1c3-a4220c519224.png)
That doesn't happen when it's installed as part of an extension pack.
